### PR TITLE
Deploy challenge services to AWS

### LINKED ISSUE #4: Deploy challenge-service to cloud service with docker-compose
## Scope

We want to run challenge-services in the cloud and make it easy for other to do so too.

## Deliverables

- Adjust challenge-services such that it can be easily deployed by a third-party user
- Verify your work by deploying a working instance into AWS

## Gain for the project

- More security

## Roles

bounty gardener: @TimDaub / 10%
bounty worker: name / 75%
bounty reviewer: name / 15%

### PR DESCRIPTION
**Pull request by TimDaub**
Fixes #4 

Note: I've also tried out AWS ECS as part of solving #4 but the effort failed: https://github.com/leapdao/challenge-services/pull/7

TODOs:

- [x] Remove redis password for easier handling
- [x] Add image to quay.io leapdao account
- [x] Remove config file from docker image and add as a volume 
- [x] Add watchtower for continous deployment (will follow up with #3)
- [x] Deploy on AWS

### CODE REVIEW

File: docker-compose-aws.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+version: "3.1"
+services:
+  event_scanner:
+    container_name: event_scanner
+    image: timdaub/challenge-services
```
Comment (TimDaub) [inline docker-compose-aws.yml]: The idea is to move this to the LeapDAO dockerhub account. Haven't gotten access yet...

**Comment by troggy** [comment]
I did the following:
1. Cloned out the repo
2. Checked out the branch for this PR
3. `npm install`
4. created and filled `config.json`
5. `npm test`

Getting the following:

```
troggy@graybox  ~/projects/acebusters/challenge-services   feat/initial-deployment ●  npm test

> challenge-services@0.0.1 test /Users/troggy/projects/acebusters/challenge-services
> mocha --recursive



  Event Scanner
    ✓ should set a block height in the database if none is defined for given contract address (690ms)

  Init
    ✓ should return initialized web3 contracts and queue names
    1) should create a queue if it doesn't exist yet


  2 passing (776ms)
  1 failing

  1) Init
       should create a queue if it doesn't exist yet:
     ReferenceError: initQueue is not defined
      at eval (eval at __get__ (src/events/init.js:99:16), <anonymous>:1:1)
      at AsyncFunction.__get__ (src/events/init.js:99:16)
      at Context.it (test/events/init.test.js:44:55)
```

File: README.md [hunk 2]
```diff
@@ -13,10 +13,6 @@ $ mv config/index_template.json config/index.json
 Fill in all the relevant information. The template file contains a rough
 documentation of the fields.
 
-This service uses a redis instance to store its local block height and uses it
```
Comment (troggy) [inline README.md]: The service still uses a redis instance to store its local block height. Too much removed here?

**Comment by troggy** [comment]
Executing `docker-compose up --build`.

Getting:

```
Building event_scanner
Step 1/12 : FROM node:latest
 ---> 4ac0e1872789
Step 2/12 : MAINTAINER tim.daubenschuetz@gmail.com
 ---> Using cache
 ---> 25f1e85208c5
Step 3/12 : RUN apt-get update && apt-get -y install cron
 ---> Using cache
 ---> 59f53bdcbaaf
Step 4/12 : COPY ./config/crontab /etc/cron.d/crontab
 ---> Using cache
 ---> ffe63e711b1d
Step 5/12 : RUN chmod 0644 /etc/cron.d/crontab
 ---> Using cache
 ---> a4b11a01facb
Step 6/12 : RUN crontab /etc/cron.d/crontab
 ---> Using cache
 ---> e8fde4fc33d4
Step 7/12 : RUN touch /var/log/cron.log
 ---> Using cache
 ---> c6642ff0775e
Step 8/12 : WORKDIR /usr/src/event-scanner
 ---> Using cache
 ---> 15a152de5267
Step 9/12 : COPY package*.json ./
 ---> Using cache
 ---> 429eb087052a
Step 10/12 : RUN npm i
 ---> Running in ab12e7fdb2b7

> sha3@1.2.3 install /usr/src/event-scanner/node_modules/sha3
> node-gyp rebuild

make: Entering directory '/usr/src/event-scanner/node_modules/sha3/build'
  CXX(target) Release/obj.target/sha3/src/addon.o
../src/addon.cpp: In static member function 'static void SHA3Hash::Init(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)':
../src/addon.cpp:83:27: error: no matching function for call to 'v8::Object::Set(v8::Local<v8::String>&, v8::Local<v8::Function>&)'
   target->Set(className, f);
                           ^
In file included from /root/.cache/node-gyp/13.0.1/include/node/node.h:63:0,
                 from ../src/addon.cpp:1:
/root/.cache/node-gyp/13.0.1/include/node/v8.h:3416:37: note: candidate: v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, v8::Local<v8::Value>, v8::Local<v8::Value>)
   V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                     ^~~
/root/.cache/node-gyp/13.0.1/include/node/v8.h:3416:37: note:   candidate expects 3 arguments, 2 provided
/root/.cache/node-gyp/13.0.1/include/node/v8.h:3419:37: note: candidate: v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, uint32_t, v8::Local<v8::Value>)
   V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                     ^~~
/root/.cache/node-gyp/13.0.1/include/node/v8.h:3419:37: note:   candidate expects 3 arguments, 2 provided
sha3.target.mk:113: recipe for target 'Release/obj.target/sha3/src/addon.o' failed
make: Leaving directory '/usr/src/event-scanner/node_modules/sha3/build'
make: *** [Release/obj.target/sha3/src/addon.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:210:5)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Linux 4.9.184-linuxkit
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/src/event-scanner/node_modules/sha3
gyp ERR! node -v v13.0.1
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sha3@1.2.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sha3@1.2.3 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-10-24T17_45_19_155Z-debug.log
ERROR: Service 'event_scanner' failed to build: The command '/bin/sh -c npm i' returned a non-zero code: 1
```

File: README.md (hunk 2 above)
Comment (TimDaub) [reply README.md]: Yes it still uses redis, but these sentences I just put here to make clear that we need a redis password. I think they're not needed anymore now...

**Comment by TimDaub** [comment]
Deprecated in favor of #9

**Pull request closed without merging**